packages/analytics/amplify_analytics_pinpoint_dart/test/analytics_legacy_native_data_test.dart

```diff
@@ -29,20 +31,24 @@ void main() {
       when(() => legacyDataProvider.getEndpointId(pinpointAppId))
           .thenAnswer((_) async => null);
 
-      final endpointIdManager = EndpointIdManager(
+      final endpointIdManager = EndpointInfoStoreManager(
         store: store,
         legacyNativeDataProvider: legacyDataProvider,
-        pinpointAppId: pinpointAppId,
       );
-      expect(await endpointIdManager.retrieveEndpointId(), isNotNull);
+      await endpointIdManager.init(pinpointAppId: pinpointAppId);
+
+      expect(
+        endpointIdManager.endpointId,
+        isNotNull,
+      );
 
       final storeVersion = await store.read(
         key: EndpointStoreKey.version.name,
       );
       expect(storeVersion, EndpointStoreVersion.v1.name);
 
       final migratedEndpointId = await store.read(
-        key: EndpointStoreKey.endpointId.name,
+        key: pinpointAppId + EndpointStoreKey.endpointId.name,
```

Comment:
nit: string template

Reply:
Is there a way to use string templates in Dart? 
https://stackoverflow.com/questions/57230201/generic-template-string-like-in-python-in-dart

Also, not sure I want to replace this with method indirection.  

Lastly, I don't think that using string interpolation is good here.  In fact it looks a bit more confusing: 

```
 key: '$pinpointAppId${EndpointStoreKey.endpointId.name}',
 
 key: pinpointAppId + EndpointStoreKey.endpointId.name,

```